apps/chat/src/store/marketplace/marketplace.epics.ts

```diff
@@ -0,0 +1,172 @@
+import { EMPTY, Observable, concat, filter, of, switchMap } from 'rxjs';
+
+import { AnyAction } from '@reduxjs/toolkit';
+
+import { combineEpics } from 'redux-observable';
+
+import { EntityType } from '@/src/types/common';
+import { AppEpic } from '@/src/types/store';
+
+import {
+  ENTITY_TYPES,
+  FilterTypes,
+  MarketplaceQueryParams,
+  MarketplaceTabs,
+} from '@/src/constants/marketplace';
+
+import { ModelsSelectors } from '../models/models.reducers';
+import { UIActions } from '../ui/ui.reducers';
+import { MarketplaceActions } from './marketplace.reducers';
+import {
+  selectDetailsModel,
+  selectSearchTerm,
+  selectSelectedFilters,
+  selectSelectedTab,
+} from './marketplace.selectors';
+
+import { ParsedUrlQueryInput, parse } from 'querystring';
+
+const addToQuery = (
+  query: ParsedUrlQueryInput,
+  key: string,
+  value: string | undefined,
+) => {
+  if (value !== undefined) {
+    query[key] = value;
+  } else {
+    delete query[key];
+  }
+};
+
+const setQueryParamsEpic: AppEpic = (action$, state$, { router }) =>
+  action$.pipe(
+    filter(
+      (action) =>
+        MarketplaceActions.setSelectedTab.match(action) ||
+        MarketplaceActions.setDetailsModel.match(action) ||
+        MarketplaceActions.setSelectedFilters.match(action) ||
+        MarketplaceActions.setFilters.match(action) ||
+        MarketplaceActions.setSearchTerm.match(action),
+    ),
+    switchMap(() => {
+      const state = state$.value;
+      const query = parse(window.location.search.slice(1));
+      // workspace tab
+      const selectedTab = selectSelectedTab(state);
+      addToQuery(
+        query,
+        MarketplaceQueryParams.tab,
+        selectedTab === MarketplaceTabs.MY_WORKSPACE
+          ? MarketplaceTabs.MY_WORKSPACE
+          : undefined,
+      );
+      // application link
+      const reference = selectDetailsModel(state)?.reference;
+      addToQuery(
+        query,
+        MarketplaceQueryParams.model,
+        reference ? reference : undefined,
+      );
+      // filters
+      const filters = selectSelectedFilters(state);
+      addToQuery(
+        query,
+        MarketplaceQueryParams.types,
+        filters.Type.length ? filters.Type.join(',') : undefined,
+      );
+      addToQuery(
+        query,
+        MarketplaceQueryParams.topics,
+        filters.Topics.length ? filters.Topics.join(',') : undefined,
+      );
+      // search
+      const searchTerm = selectSearchTerm(state);
+      addToQuery(
+        query,
+        MarketplaceQueryParams.search,
+        searchTerm ? searchTerm : undefined,
+      );
```

Comment:
```suggestion
      const query = parse(window.location.search.slice(1));

      const selectedTab = selectSelectedTab(state);
      const reference = selectDetailsModel(state)?.reference;
      const filters = selectSelectedFilters(state);
      const searchTerm = selectSearchTerm(state);

      const newQuery = {
        ...query,
        [MarketplaceQueryParams.tab]: selectedTab === MarketplaceTabs.MY_WORKSPACE ? MarketplaceTabs.MY_WORKSPACE : undefined,
        [MarketplaceQueryParams.model]: reference ? reference : undefined,
        [MarketplaceQueryParams.types]: filters.Type.length ? filters.Type.join(',') : undefined,
        [MarketplaceQueryParams.topics]: filters.Topics.length ? filters.Topics.join(',') : undefined,
        [MarketplaceQueryParams.search]: searchTerm ? searchTerm : undefined,
      };

      const finalQuery = omitBy(newQuery, value => value === undefined);
```

and maybe get rid of addToQuery.